### PR TITLE
Sign up I - Layout

### PR DESCRIPTION
**Pull request by Julianhm9612**
## Summary

- Added schema builder library for validations
- Added button, input, container and layout components
- Added roboto font and color variables
- Added cast utility
- Added sign up screen

## Screenshots

signup

## Trello Card

https://trello.com/c/jsRPTuUq/18-sign-up-i-layout

### CODE REVIEW

File: src/components/Input/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  border: 1px solid $red;
+
+  &:focus {
+    border: 1px solid $red;
+  }
+}
+
+.error {
+  color: $red;
+}
```
Comment (ebedoyaalzate) [inline src/components/Input/styles.module.scss]: add empty line

File: src/screens/SignUp/index.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+import logo from './assets/logo-wolox-original.png';
+
+interface User {
+  email: string;
+  firstName: string;
+  lastName: string;
+  pasword: string;
+  passwordConfirmation: string;
+  [key: string]: any;
+}
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: add this interface to types folder

File: src/screens/SignUp/index.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+function SignUp() {
+  const requiredMessage = 'Este campo es requerido';
+  const minPasword = 8;
+  const maxPassword = 16;
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: Add this constants to a constants file

File: src/screens/SignUp/index.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  });
+  const onSubmit = (data: User) => {
+    data.locale = 'en';
+    console.log({ user: { ...objectCamelToSnake(data) } });
+  };
+
+  return (
+    <Container>
+      <form className={styles.form}>
+        <img src={logo} className={styles.logo} alt="Logo original de wolox" />
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: Probably this image could be in the container component

File: src/screens/SignUp/index.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        <Input name="email" label="Email" register={register} errors={errors} />
+        <Input type="password" name="password" label="Password" register={register} errors={errors} />
+        <Input
+          type="password"
+          name="passwordConfirmation"
+          label="Confirmación de Password"
+          register={register}
+          errors={errors}
+        />
+        <Button primary label="Sign Up" onClick={handleSubmit(onSubmit)} />
```
Comment (ebedoyaalzate) [inline src/screens/SignUp/index.tsx]: change this onClick for onSubmit in the form tag

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/components/Input/styles.module.scss (hunk 1 above)
Comment (Julianhm9612) [reply src/components/Input/styles.module.scss]: Has been changed :godmode:

File: src/screens/SignUp/index.tsx (hunk 2 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: It has been added in a folder :godmode:

File: src/screens/SignUp/index.tsx (hunk 3 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: Has been changed :godmode:

File: src/screens/SignUp/index.tsx (hunk 5 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: It cost me more at the beginning to make the button type dynamic. :goberserk:

File: src/screens/SignUp/index.tsx (hunk 4 above)
Comment (Julianhm9612) [reply src/screens/SignUp/index.tsx]: You're right :suspect:

**Review by ebedoyaalzate — APPROVED** [review]

**Pull request merged**